### ISSUE TITLE
UAS Tugas 13 Machine Learning Nessa

### ISSUE THREAD

**Issue by nessa0712** [nessa0712/UAS-Machine-Learning_Nessa_InformatikaB#4]
**Tugas 13**
**Self-Organizing Maps**

Self-Organizing Maps (SOM) adalah salah satu jenis jaringan saraf tiruan (Artificial Neural Networks) yang diperkenalkan oleh Teuvo Kohonen pada tahun 1982. OSM digunakan untuk pembelajaran tanpa label, dan sering digunakan untuk visualisasi dan pengelompokan data berdimensi tinggi ke dalam ruang berdimensi rendah, yang biasanya dua dimensi.
Studi Kasus
Memperaktekkan algoritma Self Organizing Maps menggunakan 40 data historikal pelamar pekerjaan dengan status 0 untuk diterima dan 1 yang sudah diterima dan kolom lain adalah nilai IPK, GPM dan berapa kali pengalaman kerja.
1.	Memanggil dan mengintall simpson

![Image](https://github.com/user-attachments/assets/c6aeacae-2372-45c3-ac78-2303e421a9cc)
a.	StandardScaler digunakan untuk standarisasi data dalam proses preprocessing sebelum data tersebut digunakan dalam algoritma pembelajaran.
b.	!pip install simpsom digunakan untuk menginstal library simpsom.
c.	import simpsom as sps digunakan untuk memanggil library simpsom agar fungsi-fungsi yang terkait dengan Self-Organizing Maps dapat digunakan.
2.	Menambahkan dan menampilkan dataset

![Image](https://github.com/user-attachments/assets/c6b06647-1da2-4576-b44b-2780bf17b5b5)
a.	X = vlog139[['gpa','gmat','work_experience','admitted']] digunakan untuk variabel X sebagai subset dari dataset vlog139 dapat mengambil kolom dan fitur input untuk gpa, gmat, work_experience, dan admitted.
b.	y = vlog139['admitted'] digunakn untuk variabel target y hanya mengambil kolom admitted dari dataset vlog139.
c.	data = scaler.fit_transform(pd.DataFrame(X)) digunakan untuk melakukan standarisasi pada data x.
d.	labels = scaler.fit_transform(pd.DataFrame(y)) digunakan untuk melakukan standarisasi pada data y.
3.	Inisialisasi SOM

![Image](https://github.com/user-attachments/assets/2f6323b1-7352-41c5-8d1d-9026da6debf3)

![Image](https://github.com/user-attachments/assets/368e8957-6be5-48ee-980b-4a386339d688)

![Image](https://github.com/user-attachments/assets/33e3dd40-3f13-4aac-8c50-ab1b2fa6442e)
a.	net = sps.SOMNet(50, 50, data) digunakan untuk membuat objek SOMNet dengan ukuran 50 neuron secara horizontal dan 50 neuron secara vertikal.
b.	train_algo = 'batch' digunakn untuk menyimpan jenis algoritma pelatihan yang digunakan dalam variabel train_algo.
c.	epochs = 1000 digunakan untuk menyimpan jumlah epoch (iterasi) yang akan digunakan untuk melatih SOM.
 
d.	start_learning_rate = 0.0 untuk menyimpan nilai learning rate awal.
e.	net.nodes_graph() menunjukan grafik peta SOM yang menunjukkan distribusi dan organisasi neuron-neuron pada grid.
f.	net.diff_graph() digunakan untuk menampilkan grafik perbedaan antar neuron pada peta SOM.
g.	net.project(data, labels=labels) digunakan untuk memetakan data ke peta SOM net.cluster(data) digunakan untuk klasterisasi.
